project/src/components/app/app.jsx

```diff
@@ -8,16 +8,27 @@ import FavoritesPage from '../favorites-page/favorites-page.jsx';
 import RoomPage from '../room-page/room-page.jsx';
 import HotFoundPage from '../not-found-page/not-found-page.jsx';
 import offerType from '../offers-prop/offers-prop.js';
-
+import cityType from '../city-prop/city-prop.js';
 function App(props) {
-  const { rentalOfferCout, offers } = props;
+  const { rentalOfferCout, offers, defaultCity } = props;
+  const [selectedOffer, setSelectedOffer] = useState({});
+
+  const onOfferHover = (offerId) => {
+    const currentOffer = offers.find((offer) =>
+      offer.id === offerId,
+    );
```

Comment:
```suggestion
    const currentOffer = offers.find((offer) =>
      offer.id === offerId,
    );
```

Вот это можно вынести в функцию `getOfferById(offer, offerId)`

И вынести этот функционал вообще из компонента, тк он с ним не связан. И так компонент будет чище